src/core/varinfo.jl

```diff
@@ -2,21 +2,28 @@ import Base.string, Base.isequal, Base.==, Base.convert
 import Base.getindex, Base.setindex!
 import Base.rand, Base.show
 
-########## VarName ##########
-
+###########
+# VarName #
+###########
 immutable VarName
   csym      ::    Symbol        # symbol generated in compilation time
   sym       ::    Symbol        # variable symbol
   indexing  ::    String        # indexing
   counter   ::    Int           # counter of same {csym, uid}
 end
 
+typealias UID Tuple{Symbol,Symbol,String,Int}
```

Comment:
Perhaps we can remove `UID`, if its existence is to enable us to use VarName as a key for dictionaries. See e.g. 

http://stackoverflow.com/questions/26458149/julia-dictionary-with-tuple-keys-containing-composite-types